packages/syft/src/syft/ast/klass.py

```diff
@@ -161,6 +172,93 @@ def run_class_method(
 
         return result
 
+    def run_class_smpc_method(
+        __self: Any,
+        *args: Tuple[Any, ...],
+        **kwargs: Any,
+    ) -> object:
+        """Run remote class method on a SharePointer and get pointer to returned object.
+
+        Args:
+            *args: Args list of class method.
+            **kwargs: Keyword args of class method.
+
+        Returns:
+            Pointer to object returned by class method.
+        """
+        # relative
+        from ..core.node.common.action.smpc_action_message import (
+            MAP_FUNC_TO_NR_GENERATOR_INVOKES,
+        )
+
+        seed_id_locations = kwargs.get("seed_id_locations", None)
+        if seed_id_locations:
+            raise ValueError(
+                "There should not be any kwargs named seed_id_locations in the kwargs for MPCTensor"
+            )
+
+        seed_id_locations = secrets.randbits(64)
+        kwargs["seed_id_locations"] = str(seed_id_locations)
+        generator = np.random.default_rng(seed_id_locations)
+
+        nr_ops = MAP_FUNC_TO_NR_GENERATOR_INVOKES[attr_path_and_name.split(".")[-1]]
+        for _ in range(nr_ops):
+            generator.bytes(16)
+
+        id_at_location = UID(UUID(bytes=generator.bytes(16)))
+
+        # we want to get the return type which matches the attr_path_and_name
+        # so we ask lib_ast for the return type name that matches out
+        # attr_path_and_name and then use that to get the actual pointer klass
+        # then set the result to that pointer klass
+        return_type_name = __self.client.lib_ast.query(
+            attr_path_and_name
+        ).return_type_name
+        resolved_pointer_type = __self.client.lib_ast.query(return_type_name)
+        result = resolved_pointer_type.pointer_type(client=__self.client)
+        result.id_at_location = id_at_location
+
+        # first downcast anything primitive which is not already PyPrimitive
+        (
+            downcast_args,
+            downcast_kwargs,
+        ) = lib.python.util.downcast_args_and_kwargs(args=args, kwargs=kwargs)
+
+        # then we convert anything which isnt a pointer into a pointer
+        pointer_args, pointer_kwargs = pointerize_args_and_kwargs(
+            args=downcast_args,
+            kwargs=downcast_kwargs,
+            client=__self.client,
+            gc_enabled=False,
```

Comment:
Q: why do we disable garbage collection here also?

Reply:
We have a remote pointer and then we ask to do a remote operation -- at that point, we lose the "pointer" locally and we also send a "GarbageCollect action" -- that action might be executed before doing the actual operation (and we lose the data we want to do the operation on)